src/services/pool.ts

```diff
@@ -89,13 +89,25 @@ export const findAllRepos = async (options: FindOptions = {}, reqId?: string) =>
 
 const getRepoUrl = (baseUrl: string, prefix: string) => assertTrailingSlash(baseUrl) + assertTrailingSlash(prefix);
 
-const mapBranches = (branches: BranchEntry[], fullBaseUrl: string) =>
-  branches.map((branchEntry) => ({
+/**
+ *
+ * branch full url util,
+ * It dynamically handling url slugs for
+ * repos that have more then one version
+ */
+const setBranchFullUrl = (branchEntry: BranchEntry, fullBaseUrl: string, useUrlSlug = true) => {
+  const urlSlug = useUrlSlug && branchEntry?.usrSlug ? branchEntry.urlSlug : '';
```

Comment:
Should `branchEntry?.usrSlug` be `branchEntry.urlSlug`? 

Optional chaining here might also be unnecessary since this function is expected to be called when mapping existing `BranchEntry` objects

Reply:
@rayangler yes that is true, but the reason I had it here was because of the optional value in typescript. Typescript doesn't expect this because of the `[key]:string` type. So typescript would yell at me if if I just had `branchEntry.urlSlug`

Reply:
@rayangler good catch, yeah that was definitely a typo.